remix-astwalker/src/astWalker.ts

```diff
@@ -98,6 +98,52 @@ export class AstWalker extends EventEmitter {
     }
   }
 
+  walkFullInternal(ast: AstNode, callback?: Function | Object) {
```

Comment:
could that throw something if an AstNodeLegacy is given as input?

Reply:
Does 40168bd address your concern?

Reply:
When this library is used from typescript, typescript  will  give a type error. The runtime check is mostly for clients of the library written in Javascript.